keel-core/src/main/kotlin/com/netflix/spinnaker/keel/events/ResourceEvent.kt

```diff
@@ -55,7 +58,9 @@ import java.time.Instant
   Type(value = ResourceTaskFailed::class, name = "ResourceTaskFailed"),
   Type(value = ResourceTaskSucceeded::class, name = "ResourceTaskSucceeded")
 )
-abstract class ResourceEvent : PersistentEvent() {
+abstract class ResourceEvent(
+  open val message: String? = null
```

Comment:
@erikmunson, this should take care of:
```typescript
  // We really should not have 3 different versions of basically
  // the same field, but right now we do.
  message?: string;
  reason?: string;
  exceptionMessage?: string;
```
https://github.com/spinnaker/deck/blob/9db0ef63071b110bd98a26f6f31e5108f6c9f15c/app/scripts/modules/core/src/domain/IManagedEntity.ts#L160-L164

Reply:
Actually, since the original properties are still there, and are the values used to fill in the `message` property in the base class, deserialization should work transparently and you should start seeing the `message` field in the JSON _in addition to_ the other ones. I'll test this out to make sure.

Reply:
oooooooooooooooooo that is great. thx for clarifying.

Reply:
I'm glad you brought this up! When testing, I found I broke deserialization, not because of the `message` field (which worked as expected), but because of the new `origin` field! It's fixed now. The output looks like this:
```yaml
- type: "ResourceCheckError"
  kind: "titus/cluster@v1"
  id: "titus:cluster:titustestvpc:lpollomdtest-test"
  application: "lpollomdtest"
  timestamp: "2020-04-10T21:42:34.028Z"
  exceptionType: "com.netflix.spinnaker.keel.plugin.CannotResolveCurrentState"
  exceptionMessage: "Unable to resolve current state of titus:cluster:titustestvpc:lpollomdtest-test\
    \ due to: timeout (through reference chain: java.util.ArrayList[10326])"
  scope: "resource"
  message: "Unable to resolve current state of titus:cluster:titustestvpc:lpollomdtest-test\
    \ due to: timeout (through reference chain: java.util.ArrayList[10326])"
  origin: "SYSTEM"
  uid: "titus:cluster:titustestvpc:lpollomdtest-test"
```

I can turn the `origin` enum values to lowercase if you prefer.